zerver/actions/users.py

```diff
@@ -432,16 +441,19 @@ def do_change_user_role(
         ]
     )
 
-    do_send_user_group_members_update_event("remove_members", old_system_group, [user_profile.id])
+    do_send_user_group_members_update_event(
+        "remove_members", old_system_group, [locked_user_profile.id]
+    )
 
-    do_send_user_group_members_update_event("add_members", system_group, [user_profile.id])
+    do_send_user_group_members_update_event("add_members", system_group, [locked_user_profile.id])
 
     if UserProfile.ROLE_MEMBER in [old_value, value]:
         update_users_in_full_members_system_group(
-            user_profile.realm, [user_profile.id], acting_user=acting_user
+            locked_user_profile.realm, [locked_user_profile.id], acting_user=acting_user
         )
 
-    send_stream_events_for_role_update(user_profile, previously_accessible_streams)
+    send_stream_events_for_role_update(locked_user_profile, previously_accessible_streams)
+    user_profile.refresh_from_db()
```

Comment:
Let's add a comment just above this line explaining that we're refreshing it from the database so that the caller sees an up-to-date version of the model (at least, I assume that's the motivation).

Reply:
Yeah (as noted above)